src/test/java/com/fauna/e2e/E2EPaginationTest.java

```diff
@@ -111,4 +112,38 @@ public void query_all_flattened() {
         }
         assertEquals(50, products.size());
     }
+
+    @Test
+    public void query_statsAreTrackedForExplicitPagination() {
+        var cfg = FaunaConfig.builder()
+                .secret("secret")
+                .endpoint("http://localhost:8443")
+                .defaultStatsCollector()
+                .build();
+        var client = Fauna.client(cfg);
+        PageIterator<Product> iter = client.paginate(fql("Product.all()"), Product.class);
+        iter.forEachRemaining(page -> {});
+
+        var stats = client.getStatsCollector().get().read();
+        assertEquals(82, stats.getReadOps());
+        assertEquals(4, stats.getComputeOps());
+    }
+
+    @Test
+    public void query_statsAreTrackedForFlattenedPagination() {
+        var cfg = FaunaConfig.builder()
+                .secret("secret")
+                .endpoint("http://localhost:8443")
+                .defaultStatsCollector()
+                .build();
+        var client = Fauna.client(cfg);
+        PageIterator<Product> iter = client.paginate(fql("Product.all()"), Product.class);
+        Iterator<Product> productIter = iter.flatten();
+        for (Product p : (Iterable<Product>) () -> productIter) {
```

Comment:
This feels so wrong :P

Reply:
The API? There's still time to improve it if you want to sic a java engineer on it :)

Reply:
![98aoq7](https://github.com/user-attachments/assets/3a6ae6a2-00ae-4347-b5e7-44df86064bb8)

It's fine, just weird seeing an empty for loop like that. But Java is that way, you need to walk the iterator to empty it.